test/release/examples/primers/taskParallel.chpl

```diff
@@ -97,3 +97,72 @@ coforall i in 1..n {
 // The order of output is undefined.
 writeln("5: output from main task");
 
+/*
+.. _primers-taskparallel-task-intents:
+
+Task Intents
+------------
+
+The body of a task construct name some variables declared outside,
+or "outer variables". If so, "shadow variables" are introduced.
+Each task created by the task construct gets its own set of
+shadow variables, one per outer variable.
+
+ - Each shadow variable behaves as if it were a formal argument
+   of the task function for the task. The outer variable is passed
+   to this formal argument according to the argument intent
+   associated with the shadow variable, which is called a "task intent".
+
+ - The name of an outer variable lexically in the task construct
+   implicitly refers to the corresonding shadow variable.
+
+ - Each shadow variable is deallocated at the end of its task.
+
+The default argument intent is used by default. For numeric types,
+this implies capturing the value of the outer variable by the time
+the task starts executing. Arrays are passed by reference, and so are
+sync, single, and atomic variables. For ``begin`` statements, for example,
+this means that the captured value of an outer numeric variable
+can be accessed even after its scope exits, while an outer array variable
+cannot.
+*/
+var outerIntVariable = 2;  
+begin assert(outerIntVariable == 2);
+
+// The task intents ``in``, ``const in``, ``ref``, ``const ref``,
+// and ``reduce`` can be specified explicitly using a ``with`` clause.
+//
+// An ``in`` or ``const in`` intent creates a copy of the outer variable
+// for each task. A ``ref`` or ``const ref`` makes the
+// shadow variable an aliass for the outer variable.
+var outerArray = [10, 11, 12];
+begin with (in outerArray) assert(outerArray[1] == 11);
+
+var outerRealVariable = 1.0;
+
+coforall i in 1..n with (ref outerRealVariable) {
+  if i == 1 then
+    outerRealVariable *= 2;  // beware of potential for data races
+}
+
+// A reduce intent can be used to compute reductions with ``coforall`` loops.
+// The values of each reduce-intent shadow variable at the end of its task
+// is combined onto its outer variable according to the specified reduction
+// operation.
+var outerMaxVariable = 0;
+var outerMinVariable = 0;
+// outerIntVariable's value before the loop will be included in the sum
+
+coforall i in 1..n with (+ reduce outerIntVariable,
+                         max reduce outerMaxVariable,
+                         min reduce outerMinVariable) {
+  outerIntVariable += i;
+
+  if i % 2 == 0 then
+    outerMaxVariable = i;
+  else
+    outerMinVariable = -i;
```

Comment:
This might be more compelling if it actually used min/max to accumulate into the min/max variables rather than what feels a little arbitrary?  Could make the loop over something like -2..5 to make it somewhat interesting / remove the dependency on n.

Reply:
When the reduce intent is used with 'coforall', the shadow variable always starts at the identity at the beginning of each loop iteration. So things like `maxVar = max(maxVar, i)` are no different than `maxVar = i;`. Or maybe I did not understand your point. In any case as a synthetic example this looks OK to me. Still, I am open to suggestions.

Reply:
Oh, that's a good point, I think you're right that I was thinking of forall loops.  I think my comment still stands somewhat, though.  It seems arbitrary to me to assign even values of `i` to `outerMaxVariable` and the negation of odd values of `i` to `outerMinVariable`.  I.e., yes, this computes something, but does it resemble anything would actually compute or use these reduce intents for in practice?  I think primers are most compelling when the patterns they use to illustrate a point resonate with a user and make them think "Ah, so that's why/how I would use that."  As someone who understands the task intents concepts (but admittedly missed the forall vs. coforall aspect of this example), this example mostly makes me think "Nobody would ever do this, would they?"  Same for the `i == 1` case.  My preference would be to focus on patterns that hit that sweet spot of being easy to understand and not so artificial as to potentially muddy the point ("What do even/odd numbers have to do with computing min/max values?")